### DISCUSSION TITLE
Model Binding Fails

### DISCUSSION THREAD

**Discussion by booni3** [wire-elements/modal#304] [Q&A]
I am having issues getting the model binding to work. For example:

```html
<button wire:click='$emit("openModal", "asset.viewer", {{ json_encode(["media" => $media->id]) }})'>
Open
</button>
```

```php
class Viewer extends ModalComponent
{
    public Media $media;

    public function mount(Media $media)
    {
        $this->media = $media;
    }

...
```

Results in an error:

```bash
Cannot assign string to property App\Http\Livewire\Asset\Viewer::$media of type Domain\Asset\Models\Media
```

My Media model has a UUID primary key, hence the string error.

I can instead pass in the media_id and then find it inside the mount method, but as I understand it within the docs this should not be needed?

```html
<button wire:click='$emit("openModal", "asset.viewer", {{ json_encode(["media_id" => $media->id]) }})'>
Open
</button>
```

```php
public function mount($media_id)
{
    $this->media = Media::find($media_id);
}
```

**Comment by PhiloNL** [comment]
You can also do the following:

```php
    public string | Media $media;

    public function mount(Media $media)
    {
        $this->media = $media;
    }
```

There is a fix for the model binding but I still have to look into it: https://github.com/wire-elements/modal/pull/296

**Reply by Karl456** [reply]
Is there any update on a fix for this? Whilst `string|Media` does indeed solve the issue. If you run PHPStan, this fails and throws errors. Would be good to get that fix reviewed and merged (assuming it also fixes the PHPStan issue).

**Reply by booni3** [reply]
I believe this is a Livewire thing and not necessarily an issue with this package. I have seen the same issue using Livewire in certain scenarios without the use of this package.

**Reply by Karl456** [reply]
I've not had this issue with any Livewire component, only with a ModalComponent from this package.

**Reply by josegus** [reply]
model binding doesn't work if you use the example from the readme file.

**Reply by PhiloNL** [reply]
Model binding has now been added in the 1.0.8
